File: NPEG/Terminals/Warn.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace NPEG.Terminals
{
    [DataContract]
    public class Warn : ALeaf
    {
        public Warn()
            : base()
        {
            this.Message = String.Empty;
        }

        [DataMember]
        public String Message
        {
            get;
            set;
        }

        [DataMember]
        public Int32 Position
        {
            get;
            set;
        }
    }
}

File: Languages/npeg_javascript/Warn.js
if(!RobustHaven) var RobustHaven={};
if(!RobustHaven.Text) RobustHaven.Text={};
if(!RobustHaven.Text.Npeg) RobustHaven.Text.Npeg={};

RobustHaven.Text.Npeg.Warn = function (message, position) {
    var message = (message != undefined) ? message : '',
        position = (position != undefined) ? position : null;

    this.getMessage = function () { 	return message; };
    this.setMessage = function ( value )	{	message = value;	};

    this.getPosition = function ()	{	return position;	};
    this.setPosition = function ( value ) {	position = value;	};
}